9.0/BlazorWebAppOidcBff/BlazorWebAppOidc/Program.cs

```diff
@@ -0,0 +1,207 @@
+        // uses a scope format of the App ID URI for an AAD B2C tenant type. If your
+        // tenant is an ME-ID tenant, the App ID URI format is different:
+        // api://{CLIENT ID}, so the full scope with an API name of "Weather.Get" is:
+        // api://{CLIENT ID}/Weather.Get
+        // The {CLIENT ID} is the application (client) ID of the MinimalApiJwt app 
+        // registration.
+
+        oidcOptions.Scope.Add("https://{DIRECTORY NAME}.onmicrosoft.com/{CLIENT ID}/Weather.Get");
+        // ........................................................................
+
+        // ........................................................................
+        // The following example Authority is configured for Microsoft Entra ID
+        // and a single-tenant application registration. Set the {TENANT ID} 
+        // placeholder to the Tenant ID. The "common" Authority 
+        // https://login.microsoftonline.com/common/v2.0/ should be used 
+        // for multi-tenant apps. You can also use the "common" Authority for 
+        // single-tenant apps, but it requires a custom IssuerValidator as shown 
+        // in the comments below. 
+
+        oidcOptions.Authority = "https://login.microsoftonline.com/{TENANT ID}/v2.0/";
+        // ........................................................................
+
+        // ........................................................................
+        // Set the Client ID for the app. Set the {CLIENT ID} placeholder to
+        // the Client ID.
+
+        oidcOptions.ClientId = "{CLIENT ID}";
+        // ........................................................................
+
+        // ........................................................................
+        // Setting ResponseType to "code" configures the OIDC handler to use 
+        // authorization code flow. Implicit grants and hybrid flows are unnecessary
+        // in this mode. In a Microsoft Entra ID app registration, you don't need to 
+        // select either box for the authorization endpoint to return access tokens 
+        // or ID tokens. The OIDC handler automatically requests the appropriate 
+        // tokens using the code returned from the authorization endpoint.
+
+        oidcOptions.ResponseType = OpenIdConnectResponseType.Code;
+        // ........................................................................
+
+        // ........................................................................
+        // // Set MapInboundClaims to "false" to obtain the original claim types from 
+        // the token. Many OIDC servers use "name" and "role"/"roles" rather than 
+        // the SOAP/WS-Fed defaults in ClaimTypes. Adjust these values if your 
+        // identity provider uses different claim types.
+
+        oidcOptions.MapInboundClaims = false;
+        oidcOptions.TokenValidationParameters.NameClaimType = JwtRegisteredClaimNames.Name;
```

Comment:
Nit:

```suggestion
        oidcOptions.TokenValidationParameters.NameClaimType = "name";
```

It's weird to use `JwtRegisteredClaimNames` for "name" but not "roles". It seems unnecessarily verbose.

Reply:
I think that's just because there isn't a way to avoid the string for roles.

I'll remove this ... and check to see if we can drop a namespace.